File: BinarySearch.js
/**
 * Binary Search
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const search = (nums, target) => {
  return helper(nums, target, 0, nums.length - 1);
};

/**
 * Helper Function
 * @param {number[]} nums
 * @param {number} target
 * @param {number} left - anchor or start
 * @param {number} right - end
 */
const helper = (nums, target, left, right) => {
  if (left > right) return -1;

  const mid = Math.floor((left + right) / 2);
  const check = nums[mid];

  return {
    [true]: () => helper(nums, target, mid + 1, right),
    [check === target]: () => mid,
    [check > target]: () => helper(nums, target, left, mid - 1)
  }.true();
};

File: C++ Program/Array/Binary_search.cpp
//Binary Search
#include<iostream>
using namespace std;

int binarySearch(int arr[],int n,int key){
    int s=0;  //s=start
    int e=n;  //e=end
    while(s<=e){
        int mid=(s+e)/2;
        if(arr[mid]==key){
        return mid;
        }
        else if(arr[mid]>key){
            e=mid-1;
        }
        else{
            s=mid+1;
        }

    }
    return -1;
}
int main(){
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    int key;
    cin>>key;
    cout<<binarySearch(arr,n,key)<<endl;
}
